cmd/context/update-kubeconfig.go

```diff
@@ -35,40 +46,65 @@ func UpdateKubeconfigCMD() *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := context.Background()
 
+			// Run context command to get the Cfg into Okteto Context
 			if err := NewContextCommand().Run(ctx, &ContextOptions{}); err != nil {
 				return err
 			}
 
-			if err := ExecuteUpdateKubeconfig(); err != nil {
+			okCtx := okteto.Context()
+			kubeconfigPath := config.GetKubeconfigPath()
+			kubetokenEnabled := false
+
+			okClient, err := okteto.NewOktetoClient()
+			if err != nil {
 				return err
 			}
-
-			return nil
+			if err := okClient.Kubetoken().CheckService(okCtx.Name, okCtx.Namespace); err == nil {
+				kubetokenEnabled = true
+			}
+			return ExecuteUpdateKubeconfig(okCtx, kubeconfigPath, kubetokenEnabled)
```

Comment:
this function has different behaviors if the auth kubetoken is activated or not. Avoid using argument flags is a proposal in improving the quality of the CLI I believe. Could we refactor this logic by removing `kubetokenEnabled` arg flag and separate the behavior in different functions when updating the kubeconfig depending if kubetoken auth is enabled or not?

Reply:
i've refactored ExecuteUpdateKubeconfig in order to inject the client provider and do the operations inside. Also i added test cases for when kubetoken in enabled.